graph/DataGraph.java

```diff
@@ -652,9 +658,12 @@ private long hasEdgeTotalCount(VertexIID.Type rootTypeIID) {
 
         private long persistedVertexCount(VertexIID.Type typeIID, boolean isTransitive) {
             if (isTransitive) {
-                return persistedVertexTransitiveCount.computeIfAbsent(typeIID, iid -> {
-                    if (isRootTypeIID(typeIID)) {
-                        return bytesToLongOrZero(storage.get(vertexTransitiveCountKey(typeIID)));
+                if (isRootTypeIID(typeIID)) {
+                    return persistedVertexCount.computeIfAbsent(typeIID, iid ->
+                            bytesToLongOrZero(storage.get(vertexTransitiveCountKey(typeIID))));
+                } else {
+                    if (persistedVertexTransitiveCount.containsKey(typeIID)) {
```

Comment:
Do not call `computeIfAbsent` recursively which causes recursive update exception.